packages/ethereum-storage/src/ipfs-manager.ts

```diff
@@ -298,230 +129,58 @@ export default class IpfsManager {
-
-      // Flag if a timeout error was thrown
-      let didTimeout = false;
-
-      const getRequest = this.ipfsConnectionModule
-        .get(getRequestString, (res: any) => {
-          let data = '';
-
-          // Chunk of response data
-          res.on('data', (chunk: string) => {
-            data += chunk;
-          });
-
-          // Response data
-          res.on('end', () => {
-            let jsonData;
-            try {
-              jsonData = JSON.parse(data);
-            } catch (error) {
-              return reject(Error('Ipfs pin request response cannot be parsed into JSON format'));
-            }
-            if (!jsonData || !jsonData.Pins) {
-              return reject(Error('Ipfs pin request response has no Pins field'));
-            }
-
-            // Return the hash of the response
-            resolve(jsonData.Pins);
-          });
-
-          // Error handling
-          res.on('error', (e: string) => {
-            // If maxRetries is set, and we haven't reached maxRetries, retry the request
-            if (
-              this.errorHandlingConfig.maxRetries &&
-              retries <= this.errorHandlingConfig.maxRetries
-            ) {
-              setTimeout(
-                () => resolve(this.pin(hashes, timeout, retries + 1)),
-                this.errorHandlingConfig.delayBetweenRetries,
-              );
-            } else {
-              return reject(Error(`Ipfs pin request response error: ${e}`));
-            }
-          });
-          res.on('aborted', () => {
-            return reject(Error('Ipfs pin request response has been aborted'));
-          });
-        })
-        .on('timeout', () => {
-          didTimeout = true;
-          // explicitly abort the request
-          getRequest.abort();
-          return reject(Error('Ipfs pin request timeout'));
-        })
-        .on('abort', () => {
-          return reject(Error('Ipfs pin request has been aborted'));
-        })
-        .on('error', (e: string) => {
-          // If the error isn't a timeout, maxRetries is set, and we haven't reached maxRetries, retry the request
-          if (
-            !didTimeout &&
-            this.errorHandlingConfig.maxRetries &&
-            retries <= this.errorHandlingConfig.maxRetries
-          ) {
-            setTimeout(
-              () => resolve(this.pin(hashes, timeout, retries + 1)),
-              this.errorHandlingConfig.delayBetweenRetries,
-            );
-          } else {
-            return reject(Error(`Ipfs pin request error: ${e}`));
-          }
-        });
-
-      if (timeout || (this.ipfsConnection.timeout && this.ipfsConnection.timeout > 0)) {
-        getRequest.setTimeout(timeout || this.ipfsConnection.timeout);
+  public async pin(hashes: string[], timeout?: number): Promise<string[]> {
+    try {
+      const response = await this.axiosInstance.get(
+        `${this.IPFS_API_PIN}?arg=${hashes.join('&arg=')}`,
```

Comment:
suggestion: use axios' `params` field instead of `?arg=`

you might need to override the paramsSerializer in the axiosInstance creation

```
// `paramsSerializer` is an optional function in charge of serializing `params`
// (e.g. https://www.npmjs.com/package/qs, http://api.jquery.com/jquery.param/)
  paramsSerializer: function (params) {
    return Qs.stringify(params, {arrayFormat: 'brackets'})
  },
 ```

Reply:
done in https://github.com/RequestNetwork/requestNetwork/pull/837/commits/26f995a9be2391027a940208ef6b0a1091a08cd2